yafc-ce: init at 2.0.1

diamond-deluxe opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

New package - YAFC Community Edition, a planner and analyzer for modded Factorio. See https://github.com/shpaass/yafc-ce/ for more details.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

diamond-deluxe committed:
maintainers: add diamond-deluxe

SigmaSquadron commented on pkgs/by-name/ya/yafc-ce/package.nix:
```diff
@@ -0,0 +1,44 @@
+  lib,
+  fetchFromGitHub,
+  buildDotnetModule,
+  pkgs,
```
```suggestion
  dotnetCorePackages,
  SDL2,
  SLD2_image,
  SDL2_ttd,
```
`pkgs` should not be used inside the Nixpkgs package set.

SigmaSquadron commented on pkgs/by-name/ya/yafc-ce/package.nix:
```diff
@@ -0,0 +1,44 @@
+
+  projectFile = "Yafc/Yafc.csproj";
+  nugetDeps = ./deps.nix;
+  dotnet-sdk = pkgs.dotnetCorePackages.dotnet_8.sdk;
+  dotnet-runtime = pkgs.dotnetCorePackages.dotnet_8.runtime;
```
```suggestion
  dotnet-sdk = dotnetCorePackages.dotnet_8.sdk;
  dotnet-runtime = dotnetCorePackages.dotnet_8.runtime;
```

SigmaSquadron commented on pkgs/by-name/ya/yafc-ce/package.nix:
```diff
@@ -0,0 +1,44 @@
+  dotnet-runtime = pkgs.dotnetCorePackages.dotnet_8.runtime;
+  executables = "Yafc";
+
+  runtimeDeps = with pkgs; [
+    SDL2
+    SDL2_image
+    SDL2_ttf
+  ];
```
```suggestion
  runtimeDeps = [
    SDL2
    SDL2_image
    SDL2_ttf
  ];
```

SigmaSquadron commented on pkgs/by-name/ya/yafc-ce/package.nix:
```diff
@@ -0,0 +1,44 @@
+    SDL2_ttf
+  ];
+
+  meta = with lib; {
```
```suggestion
  meta = {
```

SigmaSquadron commented on pkgs/by-name/ya/yafc-ce/package.nix:
```diff
@@ -0,0 +1,44 @@
+
+      YAFC Community Edition is an updated and actively-maintained version of the original YAFC.
+    '';
+    platforms = lib.systems.doubles.x86_64;
```
```suggestion
    platforms = lib.platforms.x86_64;
```

SigmaSquadron commented on pkgs/by-name/ya/yafc-ce/package.nix:
```diff
@@ -0,0 +1,44 @@
+      YAFC Community Edition is an updated and actively-maintained version of the original YAFC.
+    '';
+    platforms = lib.systems.doubles.x86_64;
+    license = licenses.gpl3;
```
```suggestion
    license = lib.licenses.gpl3;
```

SigmaSquadron commented on pkgs/by-name/ya/yafc-ce/package.nix:
```diff
@@ -0,0 +1,44 @@
+    owner = "shpaass";
+    repo = pname;
+    rev = version;
+    sha256 = "TihRIRD87a1bihIM+xj+bC6MSRT/8k8oiAGquuZpWBs=";
```
```suggestion
    hash = "sha256-TihRIRD87a1bihIM+xj+bC6MSRT/8k8oiAGquuZpWBs=";
```

SigmaSquadron reviewed (changes requested):
Please squash the `yafc-ce: mark as only supported on x86_64 platforms` commit with the main `init` one.

t4ccer commented on pkgs/by-name/ya/yafc-ce/package.nix:
```diff
@@ -0,0 +1,47 @@
+
+  src = fetchFromGitHub {
+    owner = "shpaass";
+    repo = pname;
```
```suggestion
    repo = "yafc-ce";
```
https://github.com/nix-community/nixpkgs-lint/issues/21

SigmaSquadron commented on pkgs/by-name/ya/yafc-ce/package.nix:
```diff
@@ -0,0 +1,47 @@
+  nugetDeps = ./deps.nix;
+  dotnet-sdk = dotnetCorePackages.dotnet_8.sdk;
+  dotnet-runtime = dotnetCorePackages.dotnet_8.runtime;
+  executables = "Yafc";
```
```suggestion
  executables = [ "Yafc" ];
```

Other .NET packages seem to use a list of string for this attr.

SigmaSquadron commented on pkgs/by-name/ya/yafc-ce/package.nix:
```diff
@@ -0,0 +1,47 @@
+
+  projectFile = "Yafc/Yafc.csproj";
+  nugetDeps = ./deps.nix;
+  dotnet-sdk = dotnetCorePackages.dotnet_8.sdk;
+  dotnet-runtime = dotnetCorePackages.dotnet_8.runtime;
```
```suggestion
  dotnet-sdk = dotnet.sdk;
  dotnet-runtime = dotnet.runtime;
```
*very* optional: use a `let...in` statement that binds `dotnet = dotnetCorePackages.dotnet_8`. (As seen [here](https://github.com/NixOS/nixpkgs/blob/686d942635db397457e2d77471cfca1d6948b9a4/pkgs/applications/audio/tagger/default.nix#L28-L29))
If you do use a `let...in` statement, you can remove the `rec` from the derivation and inherit attributes, like [this](https://github.com/NixOS/nixpkgs/blob/6002d45a9622accea71cdcbd1be17c465c5ae756/pkgs/by-name/sw/swayest-workstyle/package.nix#L7-L18).

---

Alternatively:
```suggestion
  dotnet-sdk = dotnetCorePackages.sdk_8_0;
  dotnet-runtime = dotnetCorePackages.runtime_8_0;
```
(As seen [here](https://github.com/NixOS/nixpkgs/blob/686d942635db397457e2d77471cfca1d6948b9a4/pkgs/development/tools/marksman/default.nix#L30-L31))

This is not a blocker and could be left as is.

SigmaSquadron reviewed (approved):
Excellent work!

---

Result of `nixpkgs-review pr 335230` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>yafc-ce</li>
  </ul>
</details>

SigmaSquadron commented:
Ofborg is having a moment.

@OfBorg eval

SigmaSquadron commented:
Squash the update commit against the init commit, producing a new `yafc-ce: init at 0.9.1` commit.

diamond-deluxe committed:
yafc-ce: init at 2.0.1

Advances Factorio calculator with mod support. Community Edition of
YAFC.

https://github.com/shpaass/yafc-ce

Release: https://github.com/shpaass/yafc-ce/releases/tag/2.0.1

Co-authored-by: TheColorman <github@colorman.me>

TheColorman commented:
I've built and tested the 2.0.1 package on x86-64 linux, works like a charm. I don't have a Darwin device to test that build, but seeing as it's a simple dotnet build, I assume it'll work.

I've added the Lua library file for aarch64 builds as it wouldn't launch without it when running on arm. I've only tested arm under `binfmt.emulatedSystems`, where I had display issues that I attribute to qemu. In either case, the upstream package doesn't provide arm linux builds, so it's not officially supported anyway.

TheColorman commented:
I got curious why the x86-64 version didn't need a lua dependency, but the aarch64 version did. Turns out yafc-ce packages the liblua library files in the source repo. This includes Windows, Linux and Darwin, so I'll remove the dependency here (it would only have been used for Darwin builds, that already come prepackaged with the library file). I'm curious whether we should be packaging the lua libraries ourselves for all cases for the sake of reproducibility though.

TheColorman reviewed (approved):
Not sure I should be reviewing a PR for which I'm a co-author, but sure ¯\\\_(ツ)_/¯.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 335230`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>yafc-ce</li>
  </ul>
</details>

---

Functionality tested, no issues.

leona-ya commented on pkgs/by-name/ya/yafc-ce/package.nix:
```diff
@@ -0,0 +1,59 @@
+    hash = "sha256-qJOEdP2jRqK0E9Q4n7YgO5jyHxztb8aVOTlLDq1VvMk=";
+  };
+in
+buildDotnetModule {
```
Can you please use `finalAttrs` here so that  `pname`, `version`, `src` are not in the let-in block and `meta` has no `rec` like so:

```nix
let
  dotnet = dotnetCorePackages.dotnet_8;
in
buildDotnetModule (finalAttrs: {
  pname = "yafc-ce";
  version = "2.0.1";

  src = fetchFromGitHub {
    owner = "shpaass";
    repo = "yafc-ce";
    rev = finalAttrs.version;
    hash = "sha256-qJOEdP2jRqK0E9Q4n7YgO5jyHxztb8aVOTlLDq1VvMk=";
  };
```

leona-ya reviewed (approved):
Thanks for the MR :)
